src/main/scala/com/fulcrumgenomics/sv/tools/AggregateSvPileup.scala

```diff
@@ -0,0 +1,319 @@
+    for (pileup1 <- pileups; pileup2 <- pileups) {
+      if (pileup1.hashCode() < pileup2.hashCode() && isCluster(pileup1, pileup2, maxDist)) {
+        map.put(pileup1, map(pileup1).appended(pileup2))
+        map.put(pileup2, map(pileup2).appended(pileup1))
+      }
+    }
+    map.toMap
+  }
+
+  /** Returns true iff the two breakpoints can be clustered, i.e., they have the same left and right chromosomes and
+   * strand, and the distance between their left and right coordinates is below the threshold (inclusive) */
+  private def isCluster(bp1: BreakpointPileup, bp2: BreakpointPileup, maxDist: Int): Boolean = {
+    BreakpointContigOrientation(bp1) == BreakpointContigOrientation(bp2) &&
+      Math.abs(bp1.left_pos - bp2.left_pos) <= maxDist &&
+      Math.abs(bp1.right_pos - bp2.right_pos) <= maxDist
+  }
+
+  /**
+   * Converts a set of pairwise neighbor relationships to a set of clusters, where each cluster consists of breakpoints
+   * that can be joined into a connected path via pairwise neighbor relationships.
+   * @param pileupToNeighbors Map of pileup to its set of neighbors
+   * @return The set of pileup clusters
+   */
+  def toClusters(pileupToNeighbors: Map[BreakpointPileup, PileupGroup]): Seq[PileupGroup] = {
+    finalizeClusters(Seq(), pileupToNeighbors)
+  }
+
+  /**
+   * Similar to `toClusters()`, but picking up from a partially complete recursive call stack.
+   * @param existingClusters The set of clusters that have already been assembled
+   * @param remainingNeighborGroups The set of remaining neighbor groups, which are all separate from the existing
+   *                                clusters (don't belong with any of the existing clusters)
+   * @return The set of pileup clusters
+   */
+  @tailrec
+  private def finalizeClusters(existingClusters: Seq[PileupGroup],
+                               remainingNeighborGroups: Map[BreakpointPileup, PileupGroup]): Seq[PileupGroup] = {
+    if (remainingNeighborGroups.isEmpty) {
+      existingClusters
+    } else {
+      extractOneCluster(remainingNeighborGroups) match {
+        case (pileupGroup: PileupGroup, remaining: Map[BreakpointPileup, PileupGroup]) =>
+          finalizeClusters(existingClusters :+ pileupGroup, remaining)
+      }
+    }
+
+  }
+
+  /**
+   * Extracts a complete cluster from the map of pileups to their neighbor sets, and returns the cluster along with the
+   * reduced map of pileup to neighbor set such that no remaining pileups in the map belong in a cluster with the
+   * returned cluster.
+   * @param pileupToNeighbors Map of pileup to its set of neighbors
+   * @return (1) One complete cluster; (2) Remaining map of pileup to neighbor set, all of which are independent of the
+   *         returned cluster.
+   */
+  private def extractOneCluster(pileupToNeighbors: Map[BreakpointPileup, PileupGroup]):
+  (PileupGroup, Map[BreakpointPileup, PileupGroup]) = {
+    extractClusterFor(pileupToNeighbors.keys.head, pileupToNeighbors)
```

Comment:
Rather than just using `.head` here, how about having it preferentially pick the most supported breakpoint that we have as the next thing to build a cluster around?